### PR TITLE
Skeleton for developer guide

### PR DESCRIPTION
**Pull request by JangusRoundstone**

### CODE REVIEW

**Review by lihka1202 — APPROVED** [review]
Approved by @lihka1202

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1,38 +1,146 @@
...
+* show all cards the cardlist contains
+* add tag to a card
+* view the card by its uuid
+
+The implementation of the features are shown below: 
+
+* Adding a card
+1. When the user enters ```card add -q ... -a ...```, the input is passed to ```Parser``` class which calls ```Parser#parseCommand()```.
+2. The parser detects the keyword "card", then calls the ```Parser#CardKeywordParser()```.
+3.
```
Comment (thomasjlalba) [inline docs/DeveloperGuide.md:88]: Implementation of the card feature seems to be incomplete

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1,38 +1,146 @@
...
+
+The implementation of the features are shown below: 
+
+* Adding a card
+1. When the user enters ```card add -q ... -a ...```, the input is passed to ```Parser``` class which calls ```Parser#parseCommand()```.
+2. The parser detects the keyword "card", then calls the ```Parser#CardKeywordParser()```.
+3. 
+
+The sequence diagram below shows how this feature works:
+{UML will be added here.}
```
Comment (thomasjlalba) [inline docs/DeveloperGuide.md:91]: Would be great if there is some initial diagram (either class/sequence diagram) to help to illustrate the implementation.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1,38 +1,146 @@
...
+| Version | As a ... | I want to ...                                     | So that I could ...                                               |     
+|---------|----------|---------------------------------------------------|-------------------------------------------------------------------|
+|  v1.0   | new user	| add a card containing the questions and answers	 | store the question in the system for later revision               |
+|  v1.0	 | new user	| see a list of all cards	                         | I can see what are all the cards I have regardless of their tags  |
+|  v1.0   |	new user	| delete some cards	                               | I can remove unwanted cards                                       |
+|  v1.0	 | new user	| save cards to a local file	                      | I can reuse the cards the next time I use the software            |
+|  v1.0	 | new user	| load questions and answers from a file	          | I can use the database of flashcards                              |
+|  v1.0	 | new user	| save cards to a local file	                      | I can reuse the cards the next time I use the software            |
```
Comment (thomasjlalba) [inline docs/DeveloperGuide.md:125]: This line seems to be a repeated user story as the one in line 123

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1,38 +1,146 @@
...
+|  v1.0	 | new user	| load questions and answers from a file	          | I can use the database of flashcards                              |
+|  v2.0	 | new user	| update the content of the card	                   | I can ensure that the content of the card is up-to-date           |
+|  v2.0	 | new user	| group my cards according to my will into "Deck"	 | I can easily access all of my cards in a certain category         |
+|  v2.0	 | new user	| tag each card into one or more topics	          | I can organize my cards according to the topics associated        |
```
Comment (thomasjlalba) [inline docs/DeveloperGuide.md:129]: It seems like the tagging feature is something that may not be best suited for a new user, but instead more useful for an existing/advanced user.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -1,38 +1,146 @@
 # Developer Guide
 
+* [Acknowledgements](#acknowledgements)
```
Comment (thomasjlalba) [inline docs/DeveloperGuide.md:3]: I like the easy access to the different sections of the guide from the table of contents!

**Review by thomasjlalba — COMMENTED** [review]
The components (headers) in the Developer Guide seems to be the correct things to include, but it seems incomplete. It would be nice if there is an initial class/object/architecture/sequence Diagram to refer to so that we can identify roughly how your program runs. 

Would also be good what the cards does and what it is exactly.

Looking forward to the final version of your DG!

**Pull request merged**